src/swarm/neo/client/mixins/AllNodesRequestCore.d

```diff
@@ -0,0 +1,648 @@
+
+    Returns:
+        instance of AllNodesRequest constructed with the provided arguments
+
+*******************************************************************************/
+
+public AllNodesRequest!(Request, Connector, Disconnected, Initialiser, Handler)
+    createAllNodesRequest ( Request, Connector, Disconnected, Initialiser, Handler )
+    ( RequestOnConn.EventDispatcherAllNodes conn,
+    Request.Context* context, Connector connector, Disconnected disconnected,
+    Initialiser initialiser, Handler handler )
+{
+    return
+        AllNodesRequest!(Request, Connector, Disconnected, Initialiser, Handler)
+        (connector, disconnected, initialiser, handler, conn, context);
+}
+
+///
+unittest
+{
+    // Example request struct with RequestCore mixed in
+    struct ExampleRequest
+    {
+        mixin ExampleRequestCore!();
+    }
+
+    // (Partial) request implementation, to be instantiated from the request's
+    // handler function.
+    scope class ExampleRequestImpl
+    {
+        private RequestOnConn.EventDispatcherAllNodes conn;
+        private ExampleRequest.Context* context;
+
+        public this ( RequestOnConn.EventDispatcherAllNodes conn,
+            ExampleRequest.Context* context )
+        {
+            this.conn = conn;
+            this.context = context;
+        }
+
+        // Instantiates an AllNodesRequest with dummy delegates passed as the
+        // various policies.
+        public void run ( )
+        {
+            Initialiser initialiser;
+            auto request = createAllNodesRequest!(ExampleRequest)(
+                this.conn, this.context, &this.connect, &this.disconnected,
+                initialiser, &this.handle);
+            request.run();
+        }
+
+        // Dummy policies...
+        private bool connect ( )
+        {
+            return true;
+        }
+
+        private void disconnected ( Exception e ) { }
+
+        private struct Initialiser
+        {
+            bool initialise ( )
+            {
+                return true;
+            }
+
+            void reset ( ) { }
+        }
+
+        private void handle ( ) { }
+    }
+}
+
+/*******************************************************************************
+
+    Helper function providing the standard logic for connecting a request-on-
+    conn of an all-nodes request. To be called from the Connector policy
+    instance of an AllNodesRequest.
+
+    Suspends the specified request-on-conn until the connection is established.
+
+    Params:
+        conn = request-on-conn to suspend until the connection is established
+
+    Returns:
+        always true (indicates that the request should be initialised)
+
+*******************************************************************************/
+
+public bool allNodesRequestConnector (
+    RequestOnConn.EventDispatcherAllNodes conn )
+{
+    bool connected = false;
+    do // TODO: maybe don't need a loop, if there are no other resume codes to handle
```

Comment:
This loop is not needed, indeed. The reason is that when `waitForReconnect` returns, we get either `connected = true`, or the program aborts (see `default` clause in switch).